openedx/core/djangoapps/schedules/management/commands/tests/send_email_base.py

```diff
@@ -299,29 +307,31 @@ def test_course_end(self, has_course_ended):
             self.assertTrue(mock_schedule_send.apply_async.called)
 
     @patch.object(tasks, 'ace')
-    def test_multiple_enrollments(self, mock_ace):
+    def test_multiple_target_schedules(self, mock_ace):
         user = UserFactory.create()
-        current_day, offset, target_day = self._get_dates()
+        current_day, offset, target_day, upgrade_deadline = self._get_dates()
         num_courses = 3
         for course_index in range(num_courses):
             ScheduleFactory.create(
                 start=target_day,
-                upgrade_deadline=target_day,
+                upgrade_deadline=upgrade_deadline,
                 enrollment__course__self_paced=True,
                 enrollment__user=user,
                 enrollment__course__id=CourseKey.from_string('edX/toy/course{}'.format(course_index))
             )
 
-        course_queries = num_courses if self.has_course_queries else 0
-        expected_query_count = NUM_QUERIES_FIRST_MATCH + course_queries + NUM_QUERIES_NO_ORG_LIST
+        additional_course_queries = num_courses - 1 if self.queries_deadline_for_each_course else 0
```

Comment:
Why -1 here?

Reply:
-1 since the query for the 1st course is accounted for.  Each of the email-variants queries `coursedynamicupgradedeadlineconfiguration` at least once.  Only the ones that have`queries_deadline_for_each_course` set to `True` have _additional _ queries for the additional courses.